### ISSUE TITLE
High volume of lint report events consume excessive resources and lead to performance issues

### ISSUE THREAD

**Issue by jldec** [opral/monorepo#2327]
Capturing this as a (slightly extreme) example of the reactive api for message lint reports.

In the test scenario, 1000 messages were added directly to the file system, and then translated with the inlang cli.

`messageLintReports.getAll.subscribe()` triggers 2000 lint report events, one for each message added, and one for each message translated. 

Each of those events contains all lint reports for all messages. In the test case there were 36 languages so that means up to 36000 lint reports per event. 😳 

The process takes about a minute with (M1) cpu maxed and many gc's observed. 

![lint-reports-screenshot](https://github.com/opral/monorepo/assets/849592/a63440b0-5047-42ba-8f0a-f633e573b80f)

**Comment by jldec** [comment]
from [internal](https://discord.com/channels/897438559458430986/1212736106077102131)
> TL;DR an api change is required. The existing messageLintReports.getAll.subscribe() api is designed to produce one event on every message change, and each of those events includes all lint reports for all messages. We won't tackle that in PR 2108.

**Comment by jldec** [comment]
and more details from this [internal discord](https://discord.com/channels/897438559458430986/1212192970996981830/1212732489248804955) on feb 29.

> I took another look and ran some experiments with throttling 
>
> bottom line - there are exactly 2000 lint report events in the test because messageLintReports.getAll.subscribe()  produces an event on every message change, and each of those events includes all lint reports for all messages. 
>
> Throttling would mean changing the api not to send an event per message update, or not to send the full list of reports per event.
>
> iow there is no throttling  opportunity left which does not alter the api semantics.
Image
